packages/core/src/secretStore/secretKeyGenerator.ts

```diff
@@ -38,50 +32,15 @@ export type GeneratedKey = {
 export async function generateSecretKey({
   keyId,
   keySize = 16,
-  dbName,
-  systemCrypto: baseCrypto,
+  secretsDb,
 }: {
   /** the ID of the key to generate (if the ID already exists, then the generated key will be returned) */
   keyId: string;
   /** size of the key to generate */
   keySize?: number;
   /** name of the database that will hold the secrets */
-  dbName: string;
-  /** custom crypto primitives to use to encrypt the secret keys */
-  systemCrypto?: SecretCrypto;
+  secretsDb: EncryptedStore<any>;
 }): Promise<GeneratedKey> {
-  const systemCrypto = baseCrypto
-    ? {
-        encrypt: (value: Uint8Array) => {
-          if (baseCrypto.version === 1) {
-            const strValue = Encoder.toBase64(value).asString;
-            return baseCrypto.encrypt(strValue);
-          }
-          // In previous versions of the systemCrypto (prior to February 2023), encrypt took a uint8Array
-          return baseCrypto.encrypt(value);
-        },
-
-        decrypt: async (value: Uint8Array) => {
-          if (typeof baseCrypto.version === 'undefined') {
-            // In previous versions of the systemCrypto (prior to February 2023), the decrypt function returned a Uint8Array
-            return baseCrypto.decrypt(value);
-          }
-          const decrypted = await baseCrypto.decrypt(value);
-          if (isBase64.test(decrypted)) {
-            return Decoder.fromBase64(decrypted).asBytes;
-          }
-          // Between June 2022 and October 2022, the systemCrypto returned a string encoded in UTF-8
-          const encoder = new TextEncoder();
-
-          return encoder.encode(decrypted);
-        },
-      }
-    : undefined;
```

Comment:
Those tests for the handling of different system crypto version has been moved to the webapp (PR coming soon).
The core should not be aware of any system crypto (use for Electron).